aws-iot-thingtype/src/main/java/software/amazon/iot/thingtype/BaseHandlerStd.java

```diff
@@ -29,4 +34,17 @@ protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final CallbackContext callbackContext,
             final ProxyClient<IotClient> proxyClient,
             final Logger logger);
+
+    protected List<Tag> listTags(final ProxyClient<IotClient> proxyClient, final String arn) {
+        String nextToken = null;
+        List<Tag> listOfTags = new ArrayList<>();
+        do {
+            final ListTagsForResourceResponse response = proxyClient.injectCredentialsAndInvokeV2(
+                    Translator.listResourceTagsRequest(arn, nextToken),
+                    proxyClient.client()::listTagsForResource);
```

Comment:
Do we need error checking here?

Reply:
Both invocation of this method is inside the try-catch block in - ReadHandler and UpdateHandler. Also, we have the corresponding unit test cases to verify exceptions are translated correctly.